modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -800,7 +802,13 @@ public String toDefaultValue(Schema p) {
                 }
             }
             return null;
+        } else if (ModelUtils.isObjectSchema(p)) {
+            if (p.getDefault() != null) {
+                return super.toDefaultValue(p);
+            }
+            return null;
```

Comment:
@bkabrda thanks for adding the condition to handle default value for model. How is this going to work? Will the default value of a model become a string? or we need to pass the model property one by one somehow?

Reply:
This effectively creates a distinction between the default value being `null` (which would be explicit `null` for JsonNullable fields) and no default value - with the previous behaviour, the default value for referenced schemas was always `null`, which doesn't work well with JsonNullable.

This doesn't really change anything, because it will do a diff from `private MyModel foo = null;` to `private MyModel foo;`, which are equivalent. But for JsonNullable, it will correctly do either `foo = new JsonNullable<MyModel>(null)` (when the default is explicit null) or `foo = new JsonNullable<MyModel>()` (when there is no default value, thus meaning that we don't want to send explicit null by default).

Hope this makes it clearer.

Reply:
Thanks for the explanation.